server/controllers/configurationsController/configurationsController.test.js

```diff
@@ -0,0 +1,210 @@
+          new Error('Rejected error message')
+        )
+
+        await ConfigurationsController.create(request, response)
+
+        expect(response.status).toHaveBeenCalledWith(400)
+        expect(response.json).toHaveBeenCalledWith({
+          error: `Rejected error message`,
+        })
+      })
+    })
+  })
+
+  describe(ConfigurationsController.edit, () => {
+    const configAttributes = { body: { name: 'new matrix', owner: 'owl' } }
+
+    describe('When successful', () => {
+      it('sends a status of 200 and with a data payload', async () => {
+        const request = createRequestWithUser(configAttributes)
+        const response = createResponse()
+
+        request.app.locals.appDb.findOneAndUpdate.mockResolvedValueOnce({
+          value: { owner: 'owl', name: 'new matrix ' },
+        })
+
+        await ConfigurationsController.edit(request, response)
+
+        expect(response.status).toHaveBeenCalledWith(200)
+        expect(response.json).toHaveBeenCalledWith({
+          data: { owner: 'owl', name: 'new matrix ' },
+        })
+      })
+    })
+
+    describe('When not found', () => {
+      it('sends a status of 400 with a message when database fails to update configuration', async () => {
+        const request = createRequestWithUser(configAttributes)
+        const response = createResponse()
+
+        request.app.locals.appDb.findOneAndUpdate.mockResolvedValueOnce({})
+
+        await ConfigurationsController.edit(request, response)
+
+        expect(response.status).toHaveBeenCalledWith(400)
+      })
+    })
+
+    describe('When unsuccessful', () => {
+      it('sends a status of 500 with a message when database there is an error', async () => {
+        const request = createRequestWithUser(configAttributes)
+        const response = createResponse()
+
+        request.app.locals.appDb.findOneAndUpdate.mockRejectedValueOnce(
+          new Error('mocked error')
+        )
+
+        await ConfigurationsController.edit(request, response)
+
+        expect(response.status).toHaveBeenCalledWith(500)
+        expect(response.json).toHaveBeenCalledWith({
+          error: `mocked error`,
+        })
+      })
+    })
+  })
+
+  describe(ConfigurationsController.index, () => {
+    const params = { uid: 'owl' }
+
+    describe('When successful', () => {
+      it('returns a status of 200 and an array of configurations', async () => {
+        const request = createRequestWithUser(params)
+        const response = createResponse()
+
+        request.app.locals.appDb.aggregate.mockImplementationOnce(() => ({
+          toArray: () => createConfigurationArrayData(),
+        }))
+
+        await ConfigurationsController.index(request, response)
+
+        expect(response.status).toHaveBeenCalledWith(200)
+        expect(response.json).toHaveBeenCalledWith({
+          data: [
+            { name: 'config1', owner: 'owl' },
+            { name: 'config2', owner: 'eagle' },
+          ],
+        })
```

Comment:
we should be asserting that these are the objects from the DB

```js
      it('returns a status of 200 and an array of configurations', async () => {
        const request = createRequestWithUser(params)
        const response = createResponse()
        const configurations = [createConfiguration({ uid: '1' }), createConfiguration({ uid: '2' })]

        request.app.locals.appDb.aggregate.mockImplementationOnce(() => ({
          toArray: () => configurations,
        }))

        await ConfigurationsController.index(request, response)

        expect(response.status).toHaveBeenCalledWith(200)
        expect(response.json).toHaveBeenCalledWith({ data: configurations })
      })
```

Reply:
[1dcf49f](https://github.com/AMP-SCZ/dpdash/pull/305/commits/1dcf49fc62eb980f67c8ed28fed3c7a2b4109eec)
[e0d3c3d](https://github.com/AMP-SCZ/dpdash/pull/305/commits/e0d3c3d035b96a8af84e42d5a9680aee6c6ea853)